### PR TITLE
Fix staticcheck warnings

### PR DESCRIPTION
**Pull request by olivierperes**
These warnings were reported by [staticcheck](https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck).

### CODE REVIEW

File: chaincode/input.go [hunk 1]
```diff
@@ -162,14 +162,14 @@ type inputComputePlanTesttuple struct {
 
 type inputLeaderboard struct {
 	ObjectiveKey   string `validate:"omitempty,len=64,hexadecimal" json:"objectiveKey"`
-	AscendingOrder bool   `json:"ascendingOrder,required"`
+	AscendingOrder bool   `validate:"required" json:"ascendingOrder"`
```
Comment (thibaultrobert) [inline chaincode/input.go]: Be careful the `required` tag  in the lib validator.v9 checks if the field is not zero value : https://godoc.org/gopkg.in/go-playground/validator.v9#hdr-Required
So for a boolean it will enforce the `True` value whereas the json `required` tag will just ensure one value is passed, either `False` or `True`. So let's keep the json one for now.

File: chaincode/input.go [hunk 2]
```diff
@@ -162,14 +162,14 @@ type inputComputePlanTesttuple struct {
...
 }
 
 type inputPermissions struct {
 	Process inputPermission `validate:"required" json:"process"`
 }
 
 type inputPermission struct {
-	Public        bool     `json:"public,required"`
+	Public        bool     `validate:"required" json:"public"`
```
Comment (thibaultrobert) [inline chaincode/input.go]: Same here.

File: chaincode/permissions.go [hunk 3]
```diff
@@ -26,14 +27,15 @@ const (
...
+*/
 
 // Permission represents one permission based on an action type
 type Permission struct {
 	// Public is true if this permission is given to the asset's owner only and
 	// the nodes listed in AuthorizedIDs (open to all nodes if false)
 	Public bool `json:"public"`
 	// AuthorizedIDs list all authorised nodes other than the asset's owner
-	AuthorizedIDs []string `json:"authorizedIDs", omitempty`
+	AuthorizedIDs []string `json:"authorizedIDs,omitempty"`
```
Comment (thibaultrobert) [inline chaincode/permissions.go]: This change is a nice catch but it belongs in its own PR and requires to update the tests since it breaks some of them.

**Review by thibaultrobert — CHANGES_REQUESTED** [review]
Although some changes are nice catch, they are too much side-effects to this PR. Moreover it breaks the tests.

File: chaincode/input.go (hunk 1 above)
Comment (olivierperes) [reply chaincode/input.go]: OK, removed `validate:"required"` in both instances.

Also left `json:required` because removing it breaks the tests, although I don’t see that it does anything. What am I missing?

File: chaincode/permissions.go (hunk 3 above)
Comment (olivierperes) [reply chaincode/permissions.go]: OK, removed. I will submit a separate PR.

File: chaincode/input.go (hunk 2 above)
Comment (olivierperes) [reply chaincode/input.go]: Done, as above.

**Comment by Kelvin-M** [comment]
@olivierperes Can you rebase on `dev` to be able to do the review ?
@thibaultrobert  After the rebase, it seems to be good to be merged ?

**Comment by olivierperes** [comment]
@Kelvin-M Rebased.

**Comment by Kelvin-M** [comment]
Thank you, I let @thibaultrobert complete his review to see if some extra work has to be done :)

**Review by thibaultrobert — APPROVED** [review]
Thanks a lot for the contribution. I think the `Unused` sections can be removed and I invite you to do so in another PR, if you're willing of course.

**Comment by olivierperes** [comment]
> Thanks a lot for the contribution. I think the `Unused` sections can be removed and I invite you to do so in another PR, if you're willing of course.

Sure, done in #35.
